### PR TITLE
Adds support to Markdown Component for rendering Widgets (and other things...)

### PR DESCRIPTION
**Pull request by elliotBraem**
As requested by Illia in the post found [here](https://social.near.page/p/root.near/95194739).

Writing {{path/to/widget~blockHeight}} in the markdown (blockHeight is optional) will call a provided onPath function. The onPath function can then render the desired Widget/Thing via the path.

### CODE REVIEW

File: src/lib/components/Markdown.js [hunk 1]
```diff
@@ -57,6 +69,10 @@ export const Markdown = (props) => {
             </code>
           );
         },
+        thing: ({ node, ...props }) => {
+          console.log(JSON.stringify(node));
```
Comment (ilblackdragon) [inline src/lib/components/Markdown.js:73]: don't need debug output anymore, right?

**Review by ilblackdragon — COMMENTED** [review]
Awesome! Can you add an example usage syntax for adding extra props? e.g. how to specify something like `<Widget src="mob.near/widgets/Profile" accountId="root.near" />`

I'll let @evgenykuzyakov to review, just small comment from my side.

**Comment by think-in-universe** [comment]
Great work! @elliotBraem I also plan to work on the widget rendering in markdown this week, but it's awesome you have done this. 

One suggestion for passing the props to the widget, maybe we can use the query parameter format like `mob.near/widgets/Profile?accountId=root.near`.

**Comment by elliotBraem** [comment]
As per @think-in-universe 's suggestion, we can use the query format like `mob.near/widgets/Profile?accountId=root.near` for embedding widgets with props.

{{mob.near/widget/Profile?accountId=root.near}}

**Comment by elliotBraem** [comment]
Rebased to dev @evgenykuzyakov 

Yes, you can try it out here : https://everything.dev/#/efiz.near/widget/test.markdown

**Comment by elliotBraem** [comment]
I like it! And I think I prefer it as well -- you got me thinking about how claiming {{}} probably wasn't the best idea; I like to be able to write {{SOMETHING}} sometimes and I'm sure others do too. Something like bos:// is unique. Want to create a PR from your branch instead? @think-in-universe

**Comment by think-in-universe** [comment]
@elliotBraem sure. Since it's an open question, we can ask people to vote their favorite syntax, and we can support multiple formats if needed. 

yeah. I can submit my PR later today for you and Eugene to review. I will update the BOS URL regex to be more generic, so it supports not only widgets but also other standards like things.

**Comment by ilblackdragon** [comment]
Curious if we should use `near://` instead `bos://`? 
Because the idea is that you retrieve component data from NEAR blockchain.
I was thinking for when there is native browser for BOS, it would be near:// URLs and also mobile could route to it.

**Comment by think-in-universe** [comment]
@ilblackdragon Good point for the native browser and mobile app routing / deep linking. 

My current thought is that `bos://` scheme makes it easier for deep linking into BOS browser or app on desktop and mobile, just like [Slack's deep linking](https://api.slack.com/reference/deep-linking#client). It's usually harder for the desktop / mobile apps to maintain a list of schemes (e.g. `near://`, `eth://`, `cosmos://`, ...) that keep changing or growing. 

Besides making `near` and `eth` schemes by themselves, we can also make them part of BOS URL.

```
bos://<chain?>/<domain?>/<src>?<query>#<fragment>
```

- `chain` is optional, default to be `near`, can also be inferred from domain in some cases (such as `social.near`, `lens.eth`)
- `domain` is optional, default to be `social.near` (contract ID) for `near`
- illegal `chain` and `domain` will fallback to default ones.

The below URLs are valid following the above pattern:

1. bos://near/social.near/mob.near/widget/Profile?accountId=root.near  
2. bos://social.near/mob.near/widget/Profile?accountId=root.near (`chain` = `near`)
3. bos://mob.near/widget/Profile?accountId=root.near  (`chain` = `near`, `contractId` = `social.near`)
4. bos://mob.near/widget/Profile  (`chain` = `near`, `contractId` = `social.near`)
5. bos://near.org/mob.near/widget/Profile?accountId=root.near   (`chain` = `near`, `contractId` = `social.near`)
6. bos://near.social/#/mob.near/widget/Profile?accountId=root.near   (`chain` = `near`, `contractId` = `social.near`)
7. bos://eth/lens.eth/alice.lens/widget/Profile?accountId=bob.lens
8. bos://eth/0x28af365578586eD5Fd500A1Dc0a3E20Fc7b2Cffa/alice.lens/widget/Profile?accountId=bob.lens
9. bos://lens.eth/alice.lens/widget/Profile?accountId=bob.lens (`chain` = `eth`)

**Comment by think-in-universe** [comment]
@ilblackdragon another option is that we don't limit the schemes in VM, but allow components and gateway to make their own decisions, though consensus are preferred to come up a standard way. 

@elliotBraem I have submitted my updated implementation in this PR: https://github.com/NearSocial/VM/pull/90  which implements a `onURL` handler for processing URLs in the markdown. Markdown components that wants to support widget rendering can use this function to build their own solutions.

**Comment by evgenykuzyakov** [comment]
I think this PR can be deprecated now. It can be achieved with `onLink` from #132 
Example:
```jsx
const onLink = (props) => {
  // TODO: Parse `props.href`
  return <a {...props} />;
};

const text = `
[**Link1**](bos://mob.near/widget/Bla?foo=bar)
`;

return <Markdown text={text} onLink={onLink} />;
```

**Pull request closed without merging**